src/jsonpath/parser.c

```diff
@@ -143,9 +137,7 @@ static bool parse_filter_list(PARSER_PARAMS, struct ast_node* tok) {
       /* [:a] => [0:a] */
       /* [a::] => [a:0:] */
       if (slice_count > zend_hash_num_elements(tok->data.d_list.ht)) {
-        if (slice_count == 1) {
-          ht_append_long(tok->data.d_list.ht, INT_MAX);
-        } else if (slice_count == 2) {
+        if (slice_count >= 1 && slice_count <= 2) {
```

Comment:
I guess we could make this `if (slice_count == 1 || slice_count == 2) {`.

Reply:
👍